core/src/main/java/edu/wpi/grip/core/MatWrapper.java

```diff
@@ -100,6 +110,13 @@ private MatWrapper(GpuMat gpuMat) {
     this.isCpu = false;
   }
 
+  private MatWrapper(Mat cpuMat, GpuMat gpuMat) {
+    Objects.requireNonNull(cpuMat, "cpuMat");
+    Objects.requireNonNull(gpuMat, "gpuMat");
+    this.cpuMat = cpuMat;
+    this.gpuMat = gpuMat;
+  }
```

Comment:
Instead of having so many overloads for the constructor, you can instead just have all of those creation methods call to this one constructor.

Might reduce the amount of code.

Also, you can do:
```
this.cpuMat = requireNonNull(cpuMat, "cpuMat");
```

Reply:
Makes sense. It is a lot of clutter.